support/cas-server-support-ldap-core/src/main/java/org/apereo/cas/authentication/support/DefaultAccountStateHandler.java

```diff
@@ -156,14 +156,16 @@ protected void handleWarning(
             return;
         }
 
-        final ZonedDateTime expDate = DateTimeUtils.zonedDateTimeOf(warning.getExpiration());
-        final long ttl = ZonedDateTime.now(ZoneOffset.UTC).until(expDate, ChronoUnit.DAYS);
-        LOGGER.debug(
+        if(warning.getExpiration()!=null){
```

Comment:
Given what I have seen from various LDAP servers and weird behaviors, I'd keep them separate but I might just ask @dfish3r to confirm.